pkg/solana/logpoller/parser.go

```diff
@@ -1,5 +1,413 @@
+	expression string
+	err        error
+}
+
+var _ primitives.Visitor = (*pgDSLParser)(nil)
+
+func (v *pgDSLParser) Comparator(_ primitives.Comparator) {}
+
+func (v *pgDSLParser) Block(prim primitives.Block) {
+	cmp, err := cmpOpToString(prim.Operator)
+	if err != nil {
+		v.err = err
+
+		return
+	}
+
+	v.expression = fmt.Sprintf(
+		"%s %s :%s",
+		blockFieldName,
+		cmp,
+		v.args.withIndexedField(blockFieldName, prim.Block),
+	)
+}
+
+func (v *pgDSLParser) Confidence(prim primitives.Confidence) {
+	switch prim.ConfidenceLevel {
+	case primitives.Finalized, primitives.Unconfirmed:
+		// solana LogPoller will only use and store finalized logs
+		// to ensure x-chain compatibility, do nothing and return no error
+		// confidence in solana is effectively a noop
+		return
+	default:
+		// still return an error for invalid confidence levels
+		v.err = fmt.Errorf("%w: %s", ErrInvalidConfidence, prim.ConfidenceLevel)
+
+		return
+	}
+}
+
+func (v *pgDSLParser) Timestamp(prim primitives.Timestamp) {
+	cmp, err := cmpOpToString(prim.Operator)
+	if err != nil {
+		v.err = err
+
+		return
+	}
+
+	var tm int64
+	if prim.Timestamp > math.MaxInt64 {
+		tm = 0
+	} else {
+		tm = int64(prim.Timestamp)
+	}
+
+	v.expression = fmt.Sprintf(
+		"%s %s :%s",
+		timestampFieldName,
+		cmp,
+		v.args.withIndexedField(timestampFieldName, time.Unix(tm, 0)),
+	)
+}
+
+func (v *pgDSLParser) TxHash(prim primitives.TxHash) {
+	txHash, err := solana.PublicKeyFromBase58(prim.TxHash)
+	if err != nil {
+		v.err = err
+
+		return
+	}
+
+	v.expression = fmt.Sprintf(
+		"%s = :%s",
+		txHashFieldName,
+		v.args.withIndexedField(txHashFieldName, txHash),
+	)
+}
+
+func (v *pgDSLParser) VisitAddressFilter(p *addressFilter) {
+	v.expression = fmt.Sprintf(
+		"%s = :%s",
+		addressFieldName,
+		v.args.withIndexedField(addressFieldName, p.address),
+	)
+}
+
+func (v *pgDSLParser) VisitEventSigFilter(p *eventSigFilter) {
+	v.expression = fmt.Sprintf(
+		"%s = :%s",
+		eventSigFieldName,
+		v.args.withIndexedField(eventSigFieldName, p.eventSig),
+	)
+}
+
+func (v *pgDSLParser) buildQuery(
+	chainID string,
+	expressions []query.Expression,
+	limiter query.LimitAndSort,
+) (string, *queryArgs, error) {
+	// reset transient properties
+	v.args = newQueryArgs(chainID)
+	v.expression = ""
+	v.err = nil
+
+	// build the query string
+	clauses := []string{logsQuery("")}
+
+	where, err := v.whereClause(expressions, limiter)
+	if err != nil {
+		return "", nil, err
+	}
+
+	clauses = append(clauses, where)
+
+	order, err := v.orderClause(limiter)
+	if err != nil {
+		return "", nil, err
+	}
+
+	if len(order) > 0 {
+		clauses = append(clauses, order)
+	}
+
+	limit := v.limitClause(limiter)
+	if len(limit) > 0 {
+		clauses = append(clauses, limit)
+	}
+
+	return strings.Join(clauses, " "), v.args, nil
+}
+
+func (v *pgDSLParser) whereClause(expressions []query.Expression, limiter query.LimitAndSort) (string, error) {
+	segment := fmt.Sprintf("WHERE %s = :solana_chain_id", chainIDFieldName)
```

Comment:
```suggestion
	segment := fmt.Sprintf("WHERE %s = :chain_id", chainIDFieldName)
```
(see my comment in `parser_test.go` about this... probably should have left it here instead)